java/ql/lib/semmle/code/java/Type.qll

```diff
@@ -249,12 +268,26 @@ private predicate wildcardExtendsObject(Wildcard wc) {
   wc.getUpperBound().getType() instanceof TypeObject
 }
 
-private predicate hasSubtypeStar(RefType t, RefType sub) {
-  sub = t
+private predicate subtypeStarMagic1(RefType t) { t = any(Wildcard w).getUpperBound().getType() }
+
+private predicate subtypeStarMagic2(RefType t) { t = any(Wildcard w).getLowerBound().getType() }
```

Comment:
Call them `getAWildcardUpper/LowerBound`?

Reply:
We could... But they really are pieces manual magic, so I thought these names were clearer.

Reply:
Similar to `s instanceof Wildcard` above suggest naming them for what they are and using `// manual magic` to signal intent

Reply:
done